include/dlaf/eigensolver/tridiag_solver/index_manipulation.h

```diff
@@ -0,0 +1,153 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <pika/algorithm.hpp>
+#include <pika/execution.hpp>
+
+#include "dlaf/eigensolver/tridiag_solver/kernels.h"
+#include "dlaf/eigensolver/tridiag_solver/tile_collector.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/matrix/index.h"
+#include "dlaf/matrix/matrix.h"
+#include "dlaf/sender/policy.h"
+#include "dlaf/sender/when_all_lift.h"
+#include "dlaf/types.h"
+#include "dlaf/util_matrix.h"
+
+namespace dlaf::eigensolver::internal {
+
+// Calculates the problem size in the tile range [i_begin, i_end)
+inline SizeType problemSize(const SizeType i_begin, const SizeType i_end,
+                            const matrix::Distribution& distr) {
+  const SizeType nb = distr.blockSize().rows();
+  const SizeType nbr = distr.tileSize(GlobalTileIndex(i_end - 1, 0)).rows();
```

Comment:
Do you think it might be worth an `DLAF_ASSERT_HEAVY` here for checking that `i_end - 1` does not fall outside the valid range for `dist`?

Reply:
tileSize should already assert in some way... or am I wrong?